src/model/model.ts

```diff
@@ -245,6 +245,11 @@ export class Model {
     return [true, rules];
   }
 
+  // updatePolicy updates a policy from the model
+  public updatePolicy(sec: string, ptype: string, oldRule: string[], newRule: string[]): boolean {
+    return this.removePolicy(sec, ptype, oldRule) && this.addPolicy(sec, ptype, newRule);
```

Comment:
You should avoid to call the removePolicy and addPolicy to reduce time overhead, the best way is to get the policy in ast(`this.model.get(sec)?.get(ptype)`) and replace oldRule with newRule.

Reply:
@Zxilly You can see how Go Casbin implements it. I remember it's implemented by a map.

Reply:
have implement, should have better performance